interview-questions.md

```diff
@@ -7,60 +7,77 @@ Then, research the question to expand on your answer. Even if you feel you have
+                    are attached to certain keywords in a programming language; e.g., in Javascript the most powerful built-in methods are those for the Array datatype. These methods allow for certain
+                    liberties to be assumed when processing data in their respective languages, as they can aid heavily in such an endeavor.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: The "Driver" uses the keyboard to type whichever code has been agreed upon/suggested and approved by each member of the pair, while the "Navigator" is responsible for keeping the code
+                moving in the right direction through advice and suggestions.
 
-  Researched answer:
+  Researched answer: Pair programming is when two developers are partnered onto a single console/machine. The benefits of such a techique are obvious when two developers with different
+                      skillsets/backgrounds are working together, as such a pair will be able to shore up their weaknesses with each others' differing expertise. The member of the pair who is not
+                      currently typing is responsible for coming up with the core concepts for the project and directing the code to a structurally sound solution.
 
 
 
-6. What is something we did in class this week you found helpful?  
+6. What is something we did in class this week you found helpful?
 
-  Your answer:
+  Your answer: Pair programming is fun and helps me with my skills socializing in the workplace. I look forward to the point where I am working out how to do something for the first time with someone.
+                It will be interesting to see how bouncing off just one other person will help the flow and keep progress going.
 
 
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-1. Iteration:
+1. Iteration: Repetition of a set chain of events, often interacting with Javascript expressions and functions. Each time a loop iterates, it performs all its actions again.
 
-2. Higher Order Functions:
+2. Higher Order Functions: A higher order function for non-Javascript languages is a function that takes two or more arguments or a function that returns another function. In Javascript functions
+                            often accept other functions as arguments as well as return them, such as the .map() function can do when used in React, so Higher Order Functions in Javascript
+                            probably refers to such functions. They are very powerful logical tools for web development.
 
-3. Jest:
+3. Jest: Jest is a modern testing library for Javascript. Notably it works with the React framework for Javascript, making it a powerful ally for developers who use TTD.
```

Comment:
Yeah we'll be using Jest and Enzyme to test React. But Jest works with almost all .js frameworks to an extent.